client/components/data/query-domain-nameservers/index.jsx

```diff
@@ -0,0 +1,20 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { useDispatch } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { fetchNameservers } from 'calypso/state/domains/nameservers/actions';
+
+export default function QueryDomainNameservers( { domainName } ) {
+	const dispatch = useDispatch();
+
+	React.useEffect( () => {
+		dispatch( fetchNameservers( domainName ) );
+	}, [ dispatch, domainName ] );
```

Comment:
I'm not sure if linting errors will complain about this but theoretically you only have to provide dependencies which can change over time (see react docs [here](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects)).

Reply:
IIRC we're adding `dispatch` as a dependency in order to prevent infinite loops. I can't recall how this can be reproduced, but it's true that it's a common pattern to include `dispatch` as a dependency in query components. Maybe we should reconsider that? In any case, it's something unrelated to this PR, IMHO.

Reply:
Thanks for elaborating! 👍 I'm still curious about the exact background but a quick search didn't reveal anything obvious to me. It seems like it's related to `useDispatch` being a custom hook.

Reply:
The return value of `useDispatch` can change when the provider is rerendered with a different Redux store:
```js
<ReduxProvider store={ flipCoin() ? store1 : store2 }>
```
I've never seen the `store` itself change in any app, but it can happen 🙂 It would be a mistake if a lint rule didn't require `dispatch` to be a dependency.

On the other hand, the `setValue` returned from `[ value, setValue ] = useState()` is really guaranteed to be always the same and never needs to be a hook dependency.